container/container.go

```diff
@@ -463,6 +465,9 @@ func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, op
 			}
 			<-done
 		}
+		if detached {
```

Comment:
I think it should be fine to use a single error here.
A detach error is sort of like broken pipe.
There should not be a case where we need to return both a detach error and some other error.

I'll take a more in depth look later today, on mobile now.